QualifiedTeachersApi/src/QualifiedTeachersApi/DataStore/Crm/DataverseAdapter.SetIttResultForTeacher.cs

```diff
@@ -296,17 +301,42 @@ public async Task<SetIttResultForTeacherLookupResult> LookupData()
                 CacheKeys.GetTeacherStatusKey("211"),  // 211 == 'Trainee Teacher:DMS'
                 _ => _dataverseAdapter.GetTeacherStatus("211", qtsDateRequired: false));
 
+            var getEarlyYearsTeacherStatusIdTask = _dataverseAdapter._cache.GetOrCreateAsync(
```

Comment:
We don't need to do it here but all these teacher status lookups make me think we should have a more explicit 'teacher status cache' and just grab them all in one hit.